jina/drivers/cache.py

```diff
@@ -1,49 +1,88 @@
 __copyright__ = "Copyright (c) 2020 Jina AI Limited. All rights reserved."
 __license__ = "Apache-2.0"
 
+import struct
 from typing import Iterable
-
-import numpy as np
-
+import os
 from . import BaseRecursiveDriver
 from ..proto import uid
 
 if False:
     from ..proto import jina_pb2
 
 
-class UniqueDocDriver(BaseRecursiveDriver):
-    """ Remove docs that previously have seen in the cache.
-
-    :class:UniqueDocDriver: follows a simple rule: it stores all new doc.id in a cache file.
-    If a doc.id hits the cache, then it is removed from the request.
+class BloomFilterDriver(BaseRecursiveDriver):
```

Comment:
Is the plan to have 2 implementations? Bloom Filters can lead to false positives, thus some documents may not be indexed. This should at least be warned?

Reply:
add a property to show FP rate. note, this very simple naive bloomfilter, there are more advanced bloomfilters. the general direction is correct, I will leave the extensions to others

Reply:
> Is the plan to have 2 implementations? 

second thought, you are right. I will add a `set`-based checker